breathecode/activity/views.py

```diff
@@ -655,3 +656,23 @@ def get(self, request, activity_id=None, academy_id=None):
 
         serializer = ActivitySerializer(results, many=True)
         return Response(serializer.data)
+
+
+class V2AcademyActivityReportView(APIView):
+
+    @capable_of('read_activity')
+    def get(self, request, academy_id=None):
+        query = request.GET.get('query', None)
+        if query is None:
+            raise ValidationException('Query json was not provided', slug='query-not-found')
+
+        query = json.loads(query)
+        result = BigQuerySet('activity')
+
+        result = result.json_query(query)
```

Comment:
It should be better that the frontend request be so dump as possible, do you have any reason to have this method? Many things like pagination is better only gets parameters from the frontend to apply restrictions to improve the security and the performance